Adds new bioluminescent traits and mutations

nianjiilical opened the pull request:
[Preview](https://i.imgur.com/UG9U7pr.png)

Because 'puke green' shouldn't be the only light color to come out of mushrooms.

- Adds five new colored bioluminescence genes: plant_gene/trait/glow/yellow, green, blue, purple, and pink.
- Adds yellow, green and purple bioluminescence to holymelons, omega weed and moonflowers, respectively.
- Adds cherry bulbs, a pink-glowing cherry mutation.
- Adds fairygrass. Mutates from grass, glows blue, contains space drugs. Can be placed like normal grass to create glowing blue fairygrass astroturf.
- Replaces glowberry's Strong Bioluminescence with White Bioluminescence, and glowcaps' Red Electrical Glow with Red Bioluminescence, for consistency.

:cl:
add: Adds 5 new colored bioluminescence traits for plants.
add: Holy melons, omega weed and moonflower now have colored bioluminescence.
add: Adds a new bioluminescent mutation for cherries.
add: Adds a new bioluminescent mutation for grass, with matching glowing astroturf tiles.
tweak: Tweaks glowberry and glowcap bioluminescence to be consistent with other types.
/:cl:

nianjiilical committed:
Adds new bioluminescent traits and mutations

Because 'puke green' shouldn't be the only light color.

nianjiilical committed:
Adds new bioluminescent traits and mutations

Because 'puke green' shouldn't be the only light color.

nianjiilical committed:
Adds new bioluminescent traits and mutations

Because 'puke green' shouldn't be the only light color.

necromanceranne commented:
Fairygrass is going to be fun to play with.

ATH1909 commented:
Are these gonna clog up the trait pool for people who are trying to roll gaseous decomposition with strange seeds or 2u doses of mutagen?

nianjiilical commented:
shit i didnt even think of the trait pool thing

if its too much of a problem ill cut the number of colors down tomorrow, maybe change glowshrooms to a nicer green and just have red/blue/purple as other options

optimumtact commented:
ignore trait pool posters

kriskog commented:
Could make the bioluminescence pull from a color var on the plant to determine light color. Also, what about the strong bioluminescence trait?

carlarctg commented:
bloatany

ATH1909 commented:
> ignore trait pool posters

Hey, it's a legitimate concern for how this might impact the balance of botany! This PR is gonna add a net 6 additional plant trait to the games that are pretty much the same, and they're redundant enough to the point where I don't think that a botanist will even bother to extract them from strange seeds when they roll them (after all, there are glowing plants that you can extract bio-luminescence traits from in the seed vendor already) unless they're a podperson or something.

I think the "make the (strong) bio-luminescence trait's color depend on the plant that has the trait" is a much better idea than how this PR is trying to approach this. Just grab the dominant color of the plant and color the light like that (well, maybe add a check or two to keep people from spreading purple light around with non-shadow emission plants).

nianjiilical commented:
> Could make the bioluminescence pull from a color var on the plant to determine light color. Also, what about the strong bioluminescence trait?

i could probably do this, it probably wouldn't be too hard to have every plant define a hex code and find a way to call that? maybe just do that for strong bioluminescence though, so that the individual bioluminescence traits can be put on glowshrooms to fill the halls with different colors

also ill probably cut the orange and maybe pink colors since orange and yellow are similar enough

in the code red electric bioluminescence was defined as "color = COLOR_LIGHT_RED" and strong biolum was "color = null", but i wasn't sure if there was a reason for that, so i changed them to use hex codes like the rest

im going to be honest this is the first time ive fucked with byond code and i did this entire pr in 2 1/2 uninterrupted hours so someone else can probably do better but ill see what i can improve later

fludd12 commented:
Good First PR tag when

Krysonism commented:
Do we really need these new plants(sans fairy grass)? It seems to me they are just recolors of existing plants + glow.

The bioluminescence traits are good but I feel that for a plant to be worth including it should have a cool reagent, a cool inherent property (like combustible lemons) or some cool crafting derivatives and preferably a new sprite.

ExcessiveUseOfCobblestone commented:
you can ignore the trait pool meme, I don't care if it's harder to get gamer traits since it was put in there to be hindered by RNG. It's certainly not worth going through every plant and adding a color string to them which also limits what colors plants can emit.

If you're going to add more plants though I'd prefer if they are at least visually different from other plants (even if it's just a recolor or something). Otherwise just toss those plants and the new color traits can also be available in the traitpool.

optimumtact commented:
claiming every plant needs a unique reagent is how botany got into the stupid situation it's in in the first place

nianjiilical committed:
Merge remote-tracking branch 'origin/bioluminescence' into bioluminescence

nianjiilical committed:
Merge remote-tracking branch 'origin/bioluminescence' into bioluminescence

nianjiilical committed:
Merge remote-tracking branch 'origin/bioluminescence' into bioluminescence

nianjiilical commented:
please bear with me as i learn how the fuck gitkraken works

changes made based on feedback:

- cut hikari-shi and the orange color gene for redundancy
- cut electric limes and moved the green bioluminescence to omega weed; this might make green harder to get than other colors, but it fits
- tweaked cherry bulb fruit/seed sprites to be more distinct from normal cherries

also added screenshots

Krysonism commented:
> 
> 
> claiming every plant needs a unique reagent is how botany got into the stupid situation it's in in the first place

You can go ahead and mark your post as off topic since i never said that.

My point was that every plant needs something interesting about them that justifies growing them.

nianjiilical has addressed my concerns with this PR.

actioninja commented on code/modules/hydroponics/plant_genes.dm:
```diff
@@ -250,14 +250,14 @@
 		if(prob(power))
-			C.electrocute_act(round(power), G, 1, SHOCK_NOGLOVES)
+			C.electrocute_act(round(power), G, 1, 1)
 
```
why the define replacement?

nianjiilical replied:
i have no idea where this change came from, this is my first time using git so i must have pulled the wrong file from somewhere? i fixed it in the below commit hopefully

nianjiilical committed:
Merge remote-tracking branch 'origin/bioluminescence' into bioluminescence
